[CI] StatelessHollowIndexShardsIT testSnapshotShardsWhileHollowing failing

**Build Scans:**
- [elasticsearch-intake #39000 / part2](https://gradle-enterprise.elastic.co/s/xdxrhuqnyhyuk)
- [elasticsearch-pull-request #142586 / part-2](https://gradle-enterprise.elastic.co/s/4tyqhjuc6jj6g)
- [elasticsearch-pull-request #142439 / part-2](https://gradle-enterprise.elastic.co/s/zlffa7lojlipo)
- [elasticsearch-pull-request #142038 / part-2](https://gradle-enterprise.elastic.co/s/qwq4jjwirzlaw)
- [elasticsearch-pull-request #141863 / part-2](https://gradle-enterprise.elastic.co/s/crpqzbcin5exe)
- [elasticsearch-pull-request #141734 / part-2](https://gradle-enterprise.elastic.co/s/pqe5cjxl7uqtc)
- [elasticsearch-pull-request #141614 / part-2](https://gradle-enterprise.elastic.co/s/4e56rhumcfk4m)
- [elasticsearch-pull-request #141622 / part-2](https://gradle-enterprise.elastic.co/s/jeg4apqamruge)
- [elasticsearch-pull-request #141567 / part-2](https://gradle-enterprise.elastic.co/s/bsgyvkq3z2f3i)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT.testSnapshotShardsWhileHollowing" -Dtests.seed=4AADC584367655A6 -Dtests.locale=lag -Dtests.timezone=Asia/Harbin -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSnapshotShardsWhileHollowing'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: [1] shard failures
```

**Issue Reasons:**
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 9 failures in test testSnapshotShardsWhileHollowing (1.0% fail rate in 913 executions)
- [main] 8 failures in step part-2 (1.6% fail rate in 512 executions)
- [main] 8 failures in pipeline elasticsearch-pull-request (1.6% fail rate in 503 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBSyntheticIdsIT.java
+++ b/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBSyntheticIdsIT.java
@@ -387,6 +387,7 @@ public class TSDBSyntheticIdsIT extends ESIntegTestCase {
         if (randomBoolean()) {
             logger.info("--> restarting the cluster");
             internalCluster().rollingRestart(new InternalTestCluster.RestartCallback());
+            ensureGreen(dataStreamName);
         } else {
             // Move all the shards to a new node to force relocations
             var newNodeName = internalCluster().startDataOnlyNode();

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -524,9 +524,6 @@ tests:
 - class: org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests
   method: testCreateAndRestoreSearchableSnapshot
   issue: https://github.com/elastic/elasticsearch/issues/148217
-- class: org.elasticsearch.datastreams.TSDBSyntheticIdsIT
-  method: testSyntheticId
-  issue: https://github.com/elastic/elasticsearch/issues/148235
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testSnapshotShardsWhileHollowing
   issue: https://github.com/elastic/elasticsearch/issues/148236

